graphene/src/layers/vector/subpath.rs

```diff
@@ -334,15 +334,14 @@ impl Subpath {
 		&mut self.0
 	}
 
-	// ** INTERFACE WITH KURBO **
-
-	// TODO Implement our own a local bounding box calculation
 	/// Return the bounding box of the shape
-	pub fn bounding_box(&self) -> Rect {
-		<&Self as Into<BezPath>>::into(self).bounding_box()
+	pub fn bounding_box(&self) -> Option<[DVec2; 2]> {
```

Comment:
We should definitely add this function to the `bezier-rs` implementation of `Subpath`. I'm wondering if it'd be beneficial to do it as part of this PR.

If you were to add this code into `libraries/bezier-rs/subpath/solvers.rs` and use the `bezier-rs` `Subpath` instead, I think you'd be able to avoid needing `PathIter` for this part of the code.

Something similar can be said for the projection part in `closest_segement`.

Reply:
Heap allocating the `bezier_rs` subpath again based on the graphite subpath every time we want the bounding box (every frame when you drag the select cage) is probably quite expensive, especially for text.

Reply:
Is there any chance we could work towards a goal of making the bezier-rs library all stack-based? Or is there no way to achieve that because of some fundamental contradiction with the problem it solves?

Reply:
The way I'm using it here (with the `PathIter`) is all stack based - this bounding_box function will not heap allocate anything.